packages/passport/sdk/src/Passport.ts

```diff
@@ -299,4 +302,55 @@ export class Passport {
     const getUserInfoResult = await this.multiRollupApiClients.passportProfileApi.getUserInfo({ headers });
     return getUserInfoResult.data.linked_addresses;
   }
+
+  public async linkExternalWallet(params: LinkWalletParams): Promise<LinkedWallet> {
+    track('passport', 'linkWallet', { type: params.type });
+
+    const user = await this.authManager.getUser();
+    if (!user) {
+      throw new PassportError('User is not logged in', PassportErrorType.NOT_LOGGED_IN_ERROR);
+    }
+
+    const isRegisteredWithIMX = isUserImx(user);
+    const isRegisteredWithZkEvm = isUserZkEvm(user);
+    if (!isRegisteredWithIMX && !isRegisteredWithZkEvm) {
+      throw new PassportError('User has not been registered', PassportErrorType.USER_NOT_REGISTERED_ERROR);
+    }
+
+    const headers = { Authorization: `Bearer ${user.accessToken}` };
+    const linkWalletV2Request = {
+      type: params.type,
+      wallet_address: params.walletAddress,
+      signature: params.signature,
+      nonce: params.nonce,
+    };
+
+    return await withPassportError(async () => {
+      const linkWalletV2Result = await this.multiRollupApiClients
+        .passportProfileApi.linkWalletV2({ linkWalletV2Request }, { headers });
+
+      // Throw PassportError for Bad Request.
+      if (linkWalletV2Result.status === 400) {
+        if (isAPIError(linkWalletV2Result.data)) {
+          const { code, message } = linkWalletV2Result.data;
+          switch (code) {
+            case 'ALREADY_LINKED':
+              throw new PassportError(message, PassportErrorType.LINK_WALLET_ALREADY_LINKED_ERROR);
+            case 'MAX_WALLETS_LINKED':
+              throw new PassportError(message, PassportErrorType.LINK_WALLET_MAX_WALLETS_LINKED_ERROR);
+            case 'DUPLICATE_NONCE':
+              throw new PassportError(message, PassportErrorType.LINK_WALLET_DUPLICATE_NONCE_ERROR);
+            case 'VALIDATION_ERROR':
+              throw new PassportError(message, PassportErrorType.LINK_WALLET_VALIDATION_ERROR);
+            default:
+              throw new PassportError(message, PassportErrorType.LINK_WALLET_GENERIC_ERROR);
+          }
+        } else {
+          // Handle unexpected bad request with a generic error message
+          throw new PassportError('LINK_WALLET error', PassportErrorType.LINK_WALLET_GENERIC_ERROR);
+        }
```

Comment:
Can we also use the `trackError` function from `@imtbl/metrics` when an error is thrown please? Should help us keep track of any issues in the future